test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusIntegrationTestExtension.java

```diff
@@ -154,9 +154,9 @@ private QuarkusTestExtensionState ensureStarted(ExtensionContext extensionContex
             currentJUnitTestClass = extensionContext.getRequiredTestClass();
         }
         // we reload the test resources if we changed test class and if we had or will have per-test test resources
-        boolean reloadTestResources = isNewTestClass
-                && (hasPerTestResources || QuarkusTestExtension.hasPerTestResources(extensionContext));
-        if ((state == null && !failedBoot) || wrongProfile || reloadTestResources) {
+        boolean reloadTestResources = false;
+        if ((state == null && !failedBoot) || wrongProfile || (reloadTestResources = isNewTestClass
```

Comment:
Did you really mean to set the variable in the test? That can be confusing to read.

Reply:
I do yeah. I did it like this because the variable is used later on